robot.py

```diff
@@ -171,24 +171,31 @@ def teleopPeriodic(self):
         self.sd.putNumber("joy_vy", joystick_vy)
         self.sd.putNumber("joy_vz", joystick_vz)
 
-        if joystick_vx or joystick_vy or joystick_vz:
-            self.chassis.set_inputs(
-                joystick_vx,
-                joystick_vy,
-                joystick_vz,
-                field_oriented=not self.joystick.getRawButton(6),
-            )
-        else:
-            self.chassis.set_inputs(0, 0, 0)
-
-        if joystick_hat != -1:
-            if self.intake.has_cargo:
-                constrained_angle = -constrain_angle(
-                    math.radians(joystick_hat) + math.pi
+        # Allow big stick movements from the driver to break out of an automation
+        if abs(joystick_vx) > 0.5 or abs(joystick_vy) > 0.5:
+            self.hatch_intake.done()
+            self.hatch_deposit.done()
+            self.cargo_deposit.done()
+
+        if not self.chassis.automation_running:
```

Comment:
Or another joystick input - the joystick keeps being fed into the chassis as it runs an automation.

Reply:
Those automations will overwrite the joystick input though before the chassis actually reads them.

Reply:
This all started because I found the "drive disabled" flag in the modules which was being triggered by the climbing automation.
I thought I'd just generalise it.
We do need it for climbing at the very least. It doesn't keep resending commands.